packages/typescript-estree/src/parseSettings/warnAboutTSVersion.ts

```diff
@@ -25,8 +25,7 @@ let warnedAboutTSVersion = false;
 
 export function warnAboutTSVersion(parseSettings: ParseSettings): void {
   if (!isRunningSupportedTypeScriptVersion && !warnedAboutTSVersion) {
-    const isTTY =
-      typeof process === 'undefined' ? false : process.stdout?.isTTY;
+    const isTTY = typeof process === 'undefined' ? false : process.stdout.isTTY;
```

Comment:
IIRC this is actually necessary
https://github.com/typescript-eslint/typescript-eslint/pull/1619

Reply:
🤔 I don't like adding support for something that's not possible per type definitions... `process.stdout` is supposed to exist. If some bundler removes it then they're telling a lie to packages like us. Hmm. #7896

Reply:
Yeah generally I'd say "I don't care" but the reason we accepted it was because it's prettier and they're one of our biggest consumers.

Reply:
I do ❤️ Prettier 😄